components/SpotifyInterface/SpotifyHttpLocalServer.c

```diff
@@ -0,0 +1,192 @@
+#include "SpotifyHttpLocalServer.h"
+static const char *TAG = "SpotifyTask";
+// ******************************
+HttpLocalServerParam_t HttpLocalServerLocalParam;
+
+/**
+ * @brief This function handles the first HTTPS request to Spotify and redirects the user to the authorization page.
+ * @param[in] req The HTTP request object.
+ * @return Returns ESP_OK if the request is processed successfully.
+ */
+static esp_err_t Spotify_RequestDataAccess(httpd_req_t *req)
+{
+    char *loc_url;
+    loc_url = (char *)malloc((2 * SMALL_BUF) * sizeof(char));
+    if (loc_url == NULL)
+    {
+        ESP_LOGI(TAG, "Failed to allocate memory for the array.\n\n");
+    }
+    memset(loc_url, 0x0, SMALL_BUF * 2);
+    if (((*HttpLocalServerLocalParam.status) == IDLE))
+    {
+
+        ESP_LOGI(TAG, "Starting authorization, sending request for TOKEN");
+        sprintf(loc_url, "http://accounts.spotify.com/authorize/?client_id=%s&response_type=code&redirect_uri=%s&scope=user-read-private%%20user-read-currently-playing%%20user-read-playback-state%%20user-modify-playback-state", ClientId, ReDirectUri);
+        httpd_resp_set_hdr(req, "Location", loc_url);
+        httpd_resp_set_type(req, "text/plain");
+        httpd_resp_set_status(req, "302");
+        httpd_resp_send(req, "", HTTPD_RESP_USE_STRLEN);
+    }
+    else
+    {
+        ESP_LOGW(TAG, "Spotify is already initiated");
+    }
+    free(loc_url);
+    return ESP_OK;
+}
+
+/**
+ * @brief This function handles the callback HTTPS request from Spotify and processes the response data.
+ * @param[in] req The HTTP request object.
+ * @return Returns ESP_OK if the request is processed successfully.
+ */
+static esp_err_t Spotify_HttpsCallbackHandler(httpd_req_t *req)
+{
+    char Buf[500];
+    if (httpd_req_get_url_query_str(req, Buf, sizeof(Buf)) == ESP_OK)
+    {
+        if (Spotify_FindCode(Buf, sizeof(Buf)) == true)
+        {
+            if (xQueueSend(*(HttpLocalServerLocalParam.HttpsBufQueue), Buf, 0) == pdTRUE)
+            {
+                ESP_LOGI(TAG, "Sent data with queue");
```

Comment:
suggestion: "Data sent to server using queue"

Reply:
i fix it , naming was wrong !